lib/remote.js

```diff
@@ -3,44 +3,61 @@
 const path = require('path');
 const os = require('os');
 const fs = require('fs-extra');
+const unzip = require('unzip2');
 const config  = require('./config');
+const utils = require('./utils');
 
-// Downloads the zip file from github and extracts it to /tmp/tldr
+let request = require('request');
+
+function createTempFolder(prefix = 'tldr') {
+  let tmpdir = path.join(os.tmpdir(), prefix);
+  let target = path.join(tmpdir, utils.uniqueId());
+
+  return fs.mkdirp(target).then(() => {
+    return {
+      get path() {
+        return target;
+      },
+      remove() {
+        return fs.remove(tmpdir);
+      }
+    };
+  });
+}
+
+// Downloads the zip file from github and extracts it to random folder inside /tmp/tldr
 exports.download = () => {
-  let request = require('request');
-  let unzip = require('unzip2');
   let url = config.get().repository;
-  let target = path.join(os.tmpdir(), 'tldr');
 
-  // Empty the tmp dir
-  return fs.emptyDir(target)
-    .then(() => {
-      // Creating the extractor
-      let extractor = unzip.Extract({ path: target });
+  // Create temp folder
+  return createTempFolder().then((tempFolder) => {
```

Comment:
Perhaps you can pass the `target` as an argument to `createTempFolder`. That would make it more testable.